subprojects/docs/src/docs/release/notes.md

```diff
@@ -169,9 +116,78 @@ In addition, this behavior can be enabled from the command line for individual b
 
 More information is available in the [Java Plugin documentation for test execution](userguide/java_plugin.html#sec:test_execution).
 
-<!--
-### Example new and noteworthy
--->
+### Allow declared reasons for dependency and resolution rules
+
+In complex builds, it can become hard to interpret dependency resolution results and why a [dependency declaration](userguide/customizing_dependency_resolution_behavior.html) or a [rule](userguide/inspecting_dependencies.html#sec:dependency_declaration_reasons) was added to a build script. To improve on this situation, we extended all the corresponding APIs with the capability to define a _reason_ for each declaration or rule. 
+
+These reasons are shown in dependency insight reports and error messages if the corresponding declaration or rule influenced the resolution result. In the future, they will also be shown in build scans.
+
+    dependencies {
+        implementation('org.ow2.asm:asm:6.0') {
+            because 'we require a JDK 9 compatible bytecode generator'
+        }
+    }
+    
+### Dependency constraints for transitive dependencies
+
+With [dependency constraints](userguide/managing_transitive_dependencies.html#sec:dependency_constraints), Gradle adds a mechanism to express constraints over transitive dependencies which are used during dependency resolution. 
+
+    dependencies {
+        implementation 'org.apache.httpcomponents:httpclient'
+    }
+
+    dependencies {
+        constraints {
+            implementation('org.apache.httpcomponents:httpclient:4.5.3') {
+                because 'previous versions have a bug impacting this application'
+            }
+            implementation('commons-codec:commons-codec:1.11') {
+                because 'version 1.9 pulled from httpclient has bugs affecting this application'
+            }
+        }
+    }
+
+In the example, the version of `commons-codec` that is brought in transitively is `1.9`. With the constraint, we express that we need at least `1.11` and Gradle will now pick that version during dependency resolution.
+
+These constraints are not yet published, but will be in a future release.
+This means that they use is targeted at applications developers over library developers for now.
```

Comment:
:x: Thinking about it again. This now comes out of the blue. Lets remove these two sentences completely.

Reply:
Removed